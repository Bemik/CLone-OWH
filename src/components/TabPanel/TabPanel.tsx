import Box from "@material-ui/core/Box";

type TabPanelProps = {
  value: number;
  index: number;
};

const TabPanel: React.FC<TabPanelProps> = ({ value, index, children, ...props }) => (
  <div
    role="tabpanel"
    hidden={value !== index}
    id={`simple-tabpanel-${index}`}
    aria-labelledby={`simple-tab-${index}`}
    {...props}
    style={{ overflow: "hidden" }}
  >
    {value === index && <Box p={2}>{children}</Box>}
  </div>
);

export default TabPanel;
