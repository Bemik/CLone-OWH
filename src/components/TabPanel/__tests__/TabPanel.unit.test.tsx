import { render } from "../../../testUtils";
import TabPanel from "../TabPanel";

describe("<TabPanel/>", () => {
  it("should render component", () => {
    const { container } = render(
      <TabPanel value={0} index={0}>
        Tab Panel Content
      </TabPanel>,
    );
    expect(container).toMatchSnapshot();
  });
});
