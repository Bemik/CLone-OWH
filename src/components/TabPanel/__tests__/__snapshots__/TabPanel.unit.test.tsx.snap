// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel/> should render component 1`] = `
<div>
  <div
    aria-labelledby="simple-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
    style="overflow: hidden;"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      Tab Panel Content
    </div>
  </div>
</div>
`;
